### ISSUE TITLE
Selectable reduction strategy

### ISSUE THREAD

**Issue by keddelzz** [morganey-lang/Morganey#84]
I'd like to have the possibility to choose which reduction strategy is used to evaluate a script or a term in the repl.

In the follwing I'd like to present the ideas I have for scripts and terms. Your ideas are welcome.
## 

I suggest adding a new command to the repl, which lets you set the reduction strategy.
For example:

```
λ> :strategy normal
The reduction strategy "normal" will now be used to evaluate terms.
λ>
```
## 

As for the evaluation my idea is to add a flag to the interpreter.

**Comment by rexim** [comment]
The problem is that our Smart Church Structure Detector completely relies on the normal order.

For example, here I just hardcoded the REPL to use call-by-name instead of normal order:

``` morganey
λ> (SUCC 0)
term: (λ f . (λ x . (f (((λ f . (λ x . x)) f) x))))
λ>
```

And here you go! The result is not recognizable as a number anymore! From the call-by-name's point of view there are no beta-redices in the head position of the term so nothing to reduce. But the normal order actually goes deeper and reduces everything even under the lambda abstractions.

I'm ok with this feature only if
1. The normal order is the default strategy always;
2. We have properly documented the limitation of our Smart Detector;
3. We have some sort of a warning in the REPL about possibly broken Smart Detection when the user changes the default strategy;

**Comment by keddelzz** [comment]
I didn't think that the detection of terms will be a problem, if they're evaluated using the call-by-name reduction strategy.

Actually, I like the detection of terms.
So maybe this isn't a good idea, should I close this issue then?

**Comment by rexim** [comment]
No-no, let's keep it. It's actually interesting. I'm just thinking, is it possible to craft the `succ` operation that reduces to the normal form with call-by-name strategy? I don't actually know the answer. :) It's probably possible. And if it is, the implementation of such function is gonna be completely different. I mean in that case the detection problem is actually just a matter of using right implementations. We need to research that. So let's keep this issue opened. :)

**Comment by keddelzz** [comment]
Ok! I'll add the research label :)

**Comment by rexim** [comment]
Cool! Thanks!
